pkg/scaling/scale_handler.go

```diff
@@ -202,11 +201,20 @@ func (h *scaleHandler) checkScalers(ctx context.Context, scalableObject interfac
 	defer scalingMutex.Unlock()
 	switch obj := scalableObject.(type) {
 	case *kedav1alpha1.ScaledObject:
+		err = h.client.Get(ctx, types.NamespacedName{Name: obj.Name, Namespace: obj.Namespace}, obj)
```

Comment:
What is the reason for this change?

Reply:
It was to make sure we refresh the scaledObject/scaledJob object before we run the scalers check loop on them. I was patching the status during tests and noticed we still use the old values when checking things like lastActiveTime.

Reply:
Makes sense, that's interesting finding, thanks!